Various fixes

Xottab-DUTY opened the pull request.

Xottab-DUTY committed:
Dependency fixes

Xottab-DUTY committed:
Added Mixed configuration for some projects without it

Required for successful building

Xottab-DUTY committed:
Missing precompiled header in Mixed configuration

Xottab-DUTY committed:
Fixed wrong dependencies in Mixed configuration

Xottab-DUTY committed:
Include fix

Xottab-DUTY committed:
Another pch fix in Mixed conf.

Xottab-DUTY committed:
Remove ".dll" suffix from LoadModule calls. #78

Kaffeine commented on src/xrCore/xrCore.h:
```diff
@@ -102,7 +102,7 @@
 
 #include <time.h>
-#define MODULE_NAME "xrCore.dll"
+#define MODULE_NAME "xrCore"
```
@nitrocaster: is this safe?

nitrocaster replied:
@Kaffeine It is. You could just read the [specification on MSDN](https://msdn.microsoft.com/en-us/library/windows/desktop/ms684175%28v=vs.85%29.aspx).

Kaffeine replied:
I already read about LoadModule(), as well as GetModuleHandle() and some other functions, but I dont know all MODULE_NAME usages. May be it's not just those methods argument. I have no IDE, nor the source code ATM on this pc.

nitrocaster replied:
The only place it's used is xrCore.cpp:49:

``` cpp
GetModuleFileName(GetModuleHandle(MODULE_NAME), fn, sizeof(fn));
```

So actually `MODULE_NAME` should be either defined in xrCore.cpp or inlined on the line above.

Xottab-DUTY committed:
fixed invalid name of exported script function(thanks to AVO)

From commit:
https://github.com/avoitishin/xray-16/commit/00e48d5a23449d661924117bb5ed9e3156445d80

Xottab-DUTY committed:
I have no time to describe this commit :D

Xottab-DUTY committed:
Fixed wrong dependency in Mixed configuration (Now fixed correctly)

Xottab-DUTY committed:
Reverted xrLC_Light changes

Xottab-DUTY committed:
Merge remote-tracking branch 'refs/remotes/OpenXRay/ai_cleanup' into ai_cleanup

Xottab-DUTY committed:
Some fix on Mixed configuration

Xottab-DUTY committed:
Visual Studio Version update

Xottab-DUTY committed:
Merge remote-tracking branch 'refs/remotes/OpenXRay/ai_cleanup' into ai_cleanup
